### PR TITLE
Fetch only data that is visible in products list

### PR DESCRIPTION
**Pull request by Cloud11PL**
I want to merge this change because it:
- Doesn't fetch attributes if no columns are selected
- Doesn't fetch richText in productList

It saves, on average, 0.16 seconds compared to the query with no channels (https://github.com/saleor/saleor-dashboard/pull/1891)

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5966 Products list fetches data for columns that are hidden.](https://app.clickup.com/t/2549495/SALEOR-5966)

File: src/products/views/ProductList/ProductList.tsx [hunk 1]
```diff
@@ -300,21 +300,28 @@ export const ProductList: React.FC<ProductListProps> = ({ params }) => {
...
       .map(getAttributeIdFromColumnValue);
   }
+  const filteredColumnIds = filterColumnIds(settings.columns);
+
+  const { data, loading, refetch } = useProductListQuery({
+    displayLoader: true,
+    variables: {
+      ...queryVariables,
+      hasChannel: !!selectedChannel,
+      hasSelectedAttributes: !!filteredColumnIds.length
```
Comment (jwm0) [inline src/products/views/ProductList/ProductList.tsx:316]:
```suggestion
      hasSelectedAttributes: filteredColumnIds.length > 0
```
makes it slightly more readable when skimming through code

File: src/components/Attributes/Attributes.tsx [hunk 2]
```diff
@@ -21,14 +21,18 @@ import { defineMessages, FormattedMessage, useIntl } from "react-intl";
 import AttributeRow from "./AttributeRow";
 import { AttributeRowHandlers, VariantAttributeScope } from "./types";
 
+interface AttributeValueFragmentWithRichText extends AttributeValueFragment {
```
Comment (jwm0) [inline src/components/Attributes/Attributes.tsx:24]: At this point, it makes sense to just create a fragment that extends AttributeValueFragment and import the output type instead.

```
fragment AttributeValueDetailsFragment on AttributeValue {
        ...AttributeValueFragment
        richText
}
```

**Review by dominik-zeglen — APPROVED** [review]

File: src/fragments/attributes.ts [hunk 3]
```diff
@@ -14,14 +14,21 @@ export const attributeValueFragment = gql`
...
     dateTime
     value
   }
 `;
 
+export const attributeValueDetailsFragment = gql`
+  ${attributeValueFragment}
+  fragment AttributeValueDetailsFragment on AttributeValue {
+    ...AttributeValueFragment
+    richText
```
Comment (jwm0) [inline src/fragments/attributes.ts:28]: Use the new fragment you created instead ‼️

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/fragments/attributes.ts (hunk 3 above)
Comment (Cloud11PL) [reply src/fragments/attributes.ts:28]: It's beginning of march and I think I already won the 🏺 of the month award

**Review by jwm0 — APPROVED** [review]

**Pull request merged**